### DISCUSSION TITLE
Glyphs plugin for e107

### DISCUSSION THREAD

**Discussion by Jimmi08** [e107inc/e107#5363] [Show and tell]
Hi,
I needed to use some icon packs independently on the front theme.   Next, I wanted to add bootstrap5 default icons.    This is the result.   My time was limited, but still - it works.  

# How to use it

Just install and activate needed icons packs

![image](https://github.com/user-attachments/assets/ef28c0ae-ccb2-4fe2-8190-97ded2fb7d96)

## Where are they loaded

in admin area in any glyphs picker  (exactly like theme preferences)  - this is managed by core

in user area (frontend) in e_header.php 

## Examples of using

1. When changing theme and your content uses icons from old theme. Example: Landing Zero theme uses Ionicons, if you want to use Bootstrap 5 theme, you need to change your used icons 

2. When building theme - it is not easy to set icons in theme glyphs. Now just copy glyphs settings from glyphs.xml file for needed pack, copy needed folders and theme'glyphs work without this plugin.

## Simplicity 

If you need different icons, it is easy to create your own plugin with your own icons. Or you can add them to this plugin.  For now, I just checked all theme.xml files for using glyphicons tags.

## How it works?

glyphs.xml structure is the same as for theme.xml

After saving preferences core prefs for glyphs is changed.  The trick is using full path to glyphs files (the reason why it works)

When you change theme, you need to save them again.

## How to check it

![image](https://github.com/user-attachments/assets/f33353fa-9b95-479c-af4d-79dfd4bbfda8)

When you click on Check:

* _Actual Site preferences_: value of 'sitetheme_glyphicons' pref
* _Actual supported packs by plugin_: value of plugin preference 'packs' (checked on)
* _Available packs_: all packs from glyphs.xml

**Comment by Alex-e107nl** [comment]
Very nice @Jimmi08 !!
